### PR TITLE
Fix `MultiSegmentPacker` Python path to forward `sequence_length`, `add_start_value`, and `add_end_value` to trimming

### PR DESCRIPTION
**Pull request by laxmareddyp**
## Problem

`MultiSegmentPacker._call_python()` does not forward `sequence_length`, `add_start_value`, or `add_end_value` to `_trim_inputs_python()`, causing the Python and TF execution paths to produce different results.

- **`sequence_length`**: The Python path always trims using `self.sequence_length` (set at init), ignoring the caller-supplied override.
- **`add_start_value` / `add_end_value`**: The Python path always reserves space for both start and end tokens in the trimming budget, even when they are disabled.

The TF path (`_trim_inputs_tf`) already handles all three parameters correctly.

## Fix

- Added `sequence_length`, `add_start_value`, `add_end_value` parameters to `_trim_inputs_python()`, mirroring `_trim_inputs_tf()`.
- Updated `_call_python()` to forward these parameters to `_trim_inputs_python()`.

## Colab Notebook

[Verification gist before and after ](https://colab.sandbox.google.com/gist/laxmareddyp/d302aff0f0f4cc3e36a4382363f01a3f/multisegmentpacker-issue.ipynb)

## Checklist

- [x] I have added all the necessary unit tests for my change.
- [x] I have verified that my change does not break existing code and works with all backends (TensorFlow, JAX, and PyTorch).
- [x] My PR is based on the latest changes of the main branch (if unsure, rebase the code).
- [x] I have followed the Keras Hub [Model contribution guidelines](https://github.com/keras-team/keras-hub/blob/master/CONTRIBUTING_MODELS.md) in making these changes.
- [x] I have followed the Keras Hub [API design guidelines](https://github.com/keras-team/keras-hub/blob/master/API_DESIGN_GUIDE.md) in making these changes.
- [x] I have signed the [Contributor License Agreement](https://cla.developers.google.com/about).

### CODE REVIEW

File: keras_hub/src/layers/preprocessing/multi_segment_packer.py [hunk 1]
```diff
@@ -503,7 +510,12 @@ def _pad(x, pad_value, padding_side, sequence_length, input_type=None):
         inputs, batched = self._canonicalize_inputs_python(inputs)
         input_type = _get_type(inputs)
 
-        segments = self._trim_inputs_python(inputs)
+        segments = self._trim_inputs_python(
+            inputs,
+            sequence_length=sequence_length,
+            add_start_value=add_start_value,
+            add_end_value=add_end_value,
+        )
```
[the remark this answers is not in the document]
Comment (laxmareddyp) [reply keras_hub/src/layers/preprocessing/multi_segment_packer.py:521]: this is a pre-existing concern . _pad() already does arithmetic with sequence_length. Added int() conversion at the top of _call_python to handle this defensively.

File: keras_hub/src/layers/preprocessing/multi_segment_packer.py [hunk 2]
```diff
@@ -406,21 +406,28 @@ def _trim_inputs_waterfall_python(self, max_seq_length, segments):
...
         if self.truncate == "round_robin":
             return self._trim_inputs_round_robin_python(
-                self.sequence_length - num_special_tokens, inputs
+                sequence_length - num_special_tokens, inputs
             )
         elif self.truncate == "waterfall":
             return self._trim_inputs_waterfall_python(
-                self.sequence_length - num_special_tokens, inputs
+                sequence_length - num_special_tokens, inputs
             )
```
[the remark this answers is not in the document]
Comment (laxmareddyp) [reply keras_hub/src/layers/preprocessing/multi_segment_packer.py:431]: This is a pre-existing pattern, the TF path `_trim_inputs_tf ` does the same unclamped subtraction. This PR aligns the Python path with the TF path. Adding max(0, ...) only here would create a new divergence. If we want clamping, it should be done in both paths as a separate change.

File: keras_hub/src/layers/preprocessing/multi_segment_packer_test.py [hunk 3]
```diff
@@ -390,6 +390,64 @@ def test_config(self):
...
+            pad_value=0,
+            sep_value=[],
+            _allow_python_workflow=self._allow_python_workflow,
+        )
+        # With add_end_value=False:
+        # Budget = 7 - 1(start) - 0(no end) = 6, round_robin → 3+3.
+        # Output: [101, 1,2,3, 5,6,7]
+        token_ids, segment_ids = packer((seq1, seq2), add_end_value=False)
+        self.assertAllEqual(token_ids, [101, 1, 2, 3, 5, 6, 7])
+        self.assertAllEqual(segment_ids, [0, 0, 0, 0, 1, 1, 1])
```
[the remark this answers is not in the document]
Comment (laxmareddyp) [reply keras_hub/src/layers/preprocessing/multi_segment_packer_test.py:449]: These are pre-existing edge cases unrelated to this fix. This PR's scope is aligning the Python path with the existing TF path behavior , the 3 tests added directly cover the 3 parameters that were not being forwarded. Testing tensor inputs and negative budgets would be testing pre-existing behavior and can be addressed separately.

**Review by hertschuh — APPROVED** [review]
Thanks for the fix!
